server/service.go

```diff
@@ -438,6 +438,9 @@ func (svr *Service) RegisterWorkConn(workConn net.Conn, newMsg *msg.NewWorkConn)
 	// Check auth.
 	if err := svr.authVerifier.VerifyNewWorkConn(newMsg); err != nil {
 		xl.Warn("Invalid authentication in NewWorkConn message on run id [%s]", newMsg.RunId)
+		msg.WriteMsg(workConn, &msg.StartWorkConn{
```

Comment:
I maybe forgot to close workConn after an error occurred.

It seems better to return an error here and close workConn if err isn't nil.

```go
...
switch {
    case *msg.NewWorkConn:
        if err := svr.RegisterWorkConn(conn, m); err != nil {
            conn.Close()
        }
}
```